scripts/deposit.ts

```diff
@@ -1,96 +1,45 @@
-import { ethers } from "ethers";
 import { toWei } from "../ts/utils";
-import { BlsAccountRegistryFactory } from "../types/ethers-contracts/BlsAccountRegistryFactory";
-import { DepositManagerFactory } from "../types/ethers-contracts/DepositManagerFactory";
-import { ExampleTokenFactory } from "../types/ethers-contracts/ExampleTokenFactory";
-import { BlsAccountRegistry } from "../types/ethers-contracts/BlsAccountRegistry";
-import { TokenRegistryFactory } from "../types/ethers-contracts/TokenRegistryFactory";
-import fs from "fs";
+import { Hubble } from "../ts/hubble";
 
 const argv = require("minimist")(process.argv.slice(2), {
-    string: ["url", "tokenID", "pubkeys", "amount"]
+    string: ["url", "pubkeys", "amount"]
 });
 
 async function main() {
-    const provider = new ethers.providers.JsonRpcProvider(argv.url);
-    const signer = provider.getSigner();
+    const providerUrl = argv.url || "http://localhost:8545";
 
-    // read genesis
-    let buf = fs.readFileSync(`genesis.json`);
-    let genesis = JSON.parse(buf.toString());
-
-    let depositManager = DepositManagerFactory.connect(
-        genesis.addresses["depositManager"],
-        signer
-    );
-
-    let accountRegistry = BlsAccountRegistryFactory.connect(
-        genesis.addresses["blsAccountRegistry"],
-        signer
-    );
-
-    let exampleToken = ExampleTokenFactory.connect(
-        genesis.addresses["exampleToken"],
-        signer
-    );
-
-    let tokenRegistry = TokenRegistryFactory.connect(
-        genesis.addresses["tokenRegistry"],
-        signer
-    );
+    const hubble = Hubble.fromDefault(providerUrl);
 
     let accIDs: number[] = [];
 
     if (argv.pubkeys) {
         const pubkeys = argv.pubkeys.split(",");
-        accIDs = await registerPublicKeys(accountRegistry, pubkeys);
+        accIDs = await hubble.registerPublicKeys(pubkeys);
     }
+    const { tokenRegistry, exampleToken, depositManager } = hubble.contracts;
 
+    const tokenID = 0;
     // match token address with tokenRegistry
-    let tokenAddress = await tokenRegistry.callStatic.safeGetAddress(
-        argv.tokenID
-    );
+    let tokenAddress = await tokenRegistry.callStatic.safeGetAddress(tokenID);
 
     if (exampleToken.address != tokenAddress) {
         throw "error token address does not match";
     }
 
     // approve depositmanager for amount
-    let amount = toWei(argv.amount);
+    const amount = parseInt(argv.amount);
+    const totalAmount = accIDs.length * amount;
     let approveTx = await exampleToken.approve(
-        genesis.addresses["depositManager"],
-        toWei(argv.amount)
+        depositManager.address,
+        toWei(totalAmount.toString())
```

Comment:
To deposit 5 tokens for each 5 public keys, we need to approve 25 tokens here.

Reply:
Yeah, I fixed this locally and forgot to push, thanks!